README.md

```diff
@@ -18,6 +23,7 @@ var r = require('@ebay/retriever');
 
 // set optional logger for missing data or type mismatch with defaultValue
 r.setLogger({
+    debug: function (messageFormat, eventType, lookupPath, defaultValue) {}, // used with get() and has()
```

Comment:
I thought that removing `debug` was a good general change, and was expecting conditional logging to use `warn` like `need()`. Did you see advantages in bringing `debug` back?

Just want to clear this up before going over the full diff.

Reply:
Well since `get()` is a lenient version of `need()`, even if users conditionally logged the error cases with `get()` it wouldn't be as grave to warrant a `warn`. If it is that serious, they can just use `need()`. I think this leniency is reflected well with `get()` using `debug`. What do you think?

Reply:
I liked removing `debug` because it simplifies the usage in general. Don't need to worry about different types of logging. And users can basically debug on their own anyway. But I also think you're right, that we need to keep `get`/`has` as lenient no-frills methods.

What do you think about leaving `get`/`has` alone, and instead putting conditional logging on `need`? This way, the `need` name is still reflective of it's usage (i.e. "if this data is missing, there is a problem"), while making it more useful for conditional scenarios. 

This way we also keep the separation of:
- `need` - when you're strict about data being present in some way
- `get`/`has` - when data is optional / warnings are not needed 

In this case, it would become:
```js
r.need(model, 'data', {}, isHomePage) // will only warn when isHomePage=true
```

Reply:
Just to clarify, without the condition as the last argument, does `need()` log `debug` or not log at all. And only when the condition is set it logs `warn`?

Reply:
Hmm I meant it differently actually. It is kinda confusing now that I think about it, because it relies on `undefined` and strict booleans.

```js
r.need(model, 'data', {}) // last param is undefined, default behavior will warn
r.need(model, 'data', {}, false) // will not warn
r.need(model, 'data', {}, true) // will warn

// will warn if data is missing and isHomePage===true
// will not warn if data is missing and isHomePage===false
r.need(model, 'data', {}, isHomePage)
```